src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -46,6 +77,12 @@ public AuthorizeFilter(AuthorizationPolicy policy)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            Policy = Policy ?? await AuthorizationPolicy.CombineAsync(PolicyProvider, AuthorizeData);
+            if (Policy == null)
+            {
+                return;
```

Comment:
Not sure if this is a good idea, if someone creates their own policy provider and for some reason the policy provider returns null due to a bug in their implementation, we silently allow the request through the pipeline. I think that in this scenario, we should throw an exception or reject the authorization. I believe we did work not very long ago to throw an exception if the policy we had was null (see https://github.com/aspnet/Mvc/pull/4758/files#diff-fbf649a01b0521186bc5d75f07c20d8fR29)